packages/compass-collection/src/stores/index.ts

```diff
@@ -76,6 +88,8 @@ const appReducer = combineReducers({
   serverVersion,
   tabs,
   isDataLake,
+  stats,
```

Comment:
I see that you have `stats` as a slice now and it inherits most of the previous plugin logic (like getting updated when collection status updates), but it doesn't look like it's ever used in the collections as you are also store stats info with every tab and this is what gets passed to the header. I think we should decide which approach to take here: either have the singleton logic copied over, or keep stats data copied over to all the tabs, but then keeping them in sync when collection status changes (this is not happening at the moment, you can reproduce by opening a collection from the sidebar without opening collections list first)